pkg/compactor/compactor.go

```diff
@@ -393,7 +393,11 @@ func (c *Compactor) starting(ctx context.Context) error {
 		// users scanner depends on the ring (to check whether an user belongs
 		// to this shard or not).
 		level.Info(c.logger).Log("msg", "waiting until compactor is ACTIVE in the ring")
+
+		ctx, cancel := context.WithTimeout(ctx, c.compactorCfg.ShardingRing.StartingTimeout)
```

Comment:
We shouldn't overwrite `ctx` otherwise the timeout applies to next steps too. You should name the returned context variable differently and use it below where you want the timeout to be applied.